src/components/PaymentMethodForm/PaymentMethodFormModalRenderer/PaymentMethodFormModalRenderer.tsx

```diff
@@ -20,23 +22,23 @@ const PaymentMethodFormModalRenderer = ({
     isCreateSuccessful,
     isUpdateSuccessful,
     onResetFormState,
+    resetCreate,
+    resetUpdate,
     setIsError,
     updateError,
 }: TPaymentMethodFormModalRendererProps) => {
     const { localize } = useTranslations();
     const { hideModal, isModalOpenFor, showModal } = useModalManager();
 
     useEffect(() => {
-        if (
+        if (createError || updateError) {
+            showModal('PaymentMethodErrorModal');
+        } else if (
             (actionType === 'ADD' && (!isCreateSuccessful || !createError)) ||
```

Comment:
since the if condition is moved up, can we remove the condition for !createError?